salt/modules/virt.py

```diff
@@ -951,6 +952,48 @@ def _gen_xml(
                                                 context["cpu"]["numa"][cell_id][
                                                     k
                                                 ] = "{}='{}'".format(k, v)
+                    elif tag == "tuning":
+                        cputune = cpu.get(tag)
+                        if cputune:
+                            for tag in cputune:
+                                param = cputune.get(tag)
+                                if param:
+                                    if tag in [
+                                        "emulatorpin",
+                                        "shares",
+                                        "period",
+                                        "quota",
+                                        "global_period",
+                                        "global_quota",
+                                        "emulator_period",
+                                        "emulator_quota",
+                                        "iothread_period",
+                                        "iothread_quota",
+                                    ]:
+                                        context["cputune"][tag] = param
+                                    elif tag in [
+                                        "cachetune",
+                                        "memorytune",
+                                        "vcpusched",
+                                    ]:
+                                        for k, v in param.items():
+                                            if v:
+                                                context["cputune"][tag][k] = v
+                                    elif tag in ["vcpupin", "iothreadpin"]:
+                                        for k, v in param.items():
+                                            if v:
+                                                context["cputune"][tag][
+                                                    k
+                                                ] = "{}='{}'".format("cpuset", v)
```

Comment:
Since the `cpuset` attribute is mandatory in the XML schema, it would make the whole code simpler to move the attribute name to the jinja template and get rid of this piece of code